src/sqlfluff/dialects/dialect_clickhouse.py

```diff
@@ -303,7 +303,107 @@ class EngineSegment(BaseSegment):
         "ENGINE",
         Ref("EqualsSegment"),
         Sequence(
-            Ref("EngineFunctionSegment"),
+            Ref("TableEngineFunctionSegment"),
+            AnySetOf(
+                Sequence(
+                    "ORDER",
+                    "BY",
+                    OneOf(
+                        Ref("BracketedColumnReferenceListGrammar"),
+                        Ref("ColumnReferenceSegment"),
+                    ),
+                    optional=True,
```

Comment:
The optionals in these sequences aren't needed, as it's covered in AnySetOf

Reply:
Fixed in commit `f56765e`.